eiffel-syntax-and-usage/the-links-object.md

```diff
@@ -1,33 +1,46 @@
 # The Links Object
-The __links__ object contains trace links to other Eiffel events. These trace links by definition always reference backwards in time – it is only possible to reference an event that has already occured. The value is always a UUID or array of UUIDs, corresponding to the __meta.id__ of the target(s), on String (or String array) format.
+The __links__ object contains trace links to other Eiffel events. These trace links by definition always reference backwards in time – it is only possible to reference an event that has already occured. The value is always an object or array of objects, the syntax of which is described below. These objects contain the target event type and a UUID corresponding to the __meta.id__ of the target, on String format.
 
 Every trace link type has a set of possible source event types, and a set of possible target event types.
 
+## Link Object Members
+### type
+__Type:__ String  
+__Format:__ An event type name  
+__Required:__ Yes  
+__Description:__ The __meta.type__ of the target event.
+
+### id
+__Type:__ String  
+__Format:__ [UUID](http://tools.ietf.org/html/rfc4122)  
+__Required:__ Yes  
+__Description:__ The __meta.id__ of the target event.
+
 ## Links Members
 ### links.causes
-__Type:__ String[]  
+__Type:__ Object[]  
 __Required in:__ None  
 __Optional in:__ Any  
 __Legal targets:__ Any  
 __Description:__ Identifies one or more causes of the event occurring. SHOULD not be used in conjunction with __links.context__: individual events providing __links.causes__ within a larger context gives rise to ambiguity. It is instead recommended to let the root event of the context declare __links.causes__.  
```

Comment:
I'm also torn here. Treating the links as object can potentially be useful, but at the same time we lose a bit of "leanness". The only(?) way can be open for such a decision and remain compatible would be to have a "degenerate" link object consisting of only an id field. This allows the addition of any meta data associated with links in the future. Would it be worth it? It poses a pedagogical problem to explain this "strange" object.

"SHOULD not" -> "SHOULD NOT".

Should we define a type EventId (which in turn is defined as being a UUID) and use that throughout rather than UUID. It would increase the abstraction level somewhat, and also better indicate what the UUID actually represents.